src/graphql/customer.rs

```diff
@@ -154,34 +191,36 @@ impl CustomerMutation {
         let i = id.as_str().parse::<u32>().map_err(|_| "invalid ID")?;
         let old = old.try_into()?;
         let new = new.try_into()?;
-        let changed_networks = {
-            let store = crate::graphql::get_store(ctx).await?;
-            let mut map = store.customer_map();
-            map.update(i, &old, &new)?;
-
-            let mut hosts = vec![];
-            let mut networks = vec![];
-            let mut ip_ranges = vec![];
-            if let Some(new_networks) = new.networks {
-                let customer_id = get_customer_id_of_node(&store).ok().flatten();
-                if customer_id == Some(i) {
-                    for nn in new_networks {
-                        let v = nn.network_group;
-                        hosts.extend(v.hosts());
-                        networks.extend(v.networks());
-                        ip_ranges.extend(v.ip_ranges().to_vec());
-                    }
-                    Some(database::HostNetworkGroup::new(hosts, networks, ip_ranges))
-                } else {
-                    None
+
+        let mut changed_networks = Vec::new();
+        let store = crate::graphql::get_store(ctx).await?;
+        let mut map = store.customer_map();
+        map.update(i, &old, &new)?;
+
+        let mut hosts = vec![];
+        let mut networks = vec![];
+        let mut ip_ranges = vec![];
```

Comment:
These lines seem to be able to move into the below `if let Some(agent_keys)`.

Reply:
Additionally, could you consider using the `fold` method instead of the `for` loop?

Reply:
done